File: Acume/src/main/java/com/guavus/acume/flex/MeasureType.java
package com.guavus.acume.flex;

public enum MeasureType {
	
	BASIC, DERIVED, DERIVED_ANNOTATED, DERIVED_DYNAMIC;
	
	/**
	 * Returns trues if the measure is derived runtime,
	 * i.e. measure is evaluated using measure processor.
	 * @return
	 */
	public boolean isDerivedRuntime() {
		if(this == DERIVED || this == DERIVED_DYNAMIC) {
			return true;
		}
		return false;
	}
	
	/**
	 * Returns true if the measure is derived,
	 * i.e. measure is evaluated using annotator
	 * or measure processor.
	 * @return
	 */
	public boolean isDerived() {
		if(this == DERIVED || this == DERIVED_ANNOTATED || this == DERIVED_DYNAMIC) {
			return true;
		}
		return false;
	}

}

File: cache/src/main/scala/com/guavus/acume/cache/common/MeasureType.scala
package com.guavus.acume.cache.common

import scala.collection.JavaConversions._

/**
 * @author archit.thakur
 *
 */
object MeasureType extends Enumeration {

  val BASIC = new MeasureType()
  val DERIVED = new MeasureType()
  val DERIVED_ANNOTATED = new MeasureType()
  val DERIVED_DYNAMIC = new MeasureType()

  class MeasureType extends Val {

    def isDerivedRuntime(): Boolean = {
      if (this == DERIVED || this == DERIVED_DYNAMIC) {
        return true
      }
      false
    }

    def isDerived(): Boolean = {
      if (this == DERIVED || this == DERIVED_ANNOTATED || this == DERIVED_DYNAMIC) {
        return true
      }
      false
    }
  }

  implicit def convertValue(v: Value): MeasureType = v.asInstanceOf[MeasureType]
}
